bot/src/main/kotlin/me/melijn/bot/services/ReminderService.kt

```diff
@@ -0,0 +1,101 @@
+package me.melijn.bot.services
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock
+import me.melijn.ap.injector.Inject
+import me.melijn.bot.database.manager.MissingUserManager
+import me.melijn.bot.database.manager.ReminderManager
+import me.melijn.bot.database.manager.openPrivateChannelSafely
+import me.melijn.bot.utils.JDAUtil.awaitOrNull
+import me.melijn.bot.utils.KoinUtil
+import me.melijn.bot.utils.Log
+import me.melijn.gen.RemindersData
+import me.melijn.kordkommons.async.TaskScope
+import net.dv8tion.jda.api.entities.Message
+import net.dv8tion.jda.api.exceptions.ErrorResponseException
+import net.dv8tion.jda.api.requests.ErrorResponse
+import net.dv8tion.jda.api.sharding.ShardManager
+import kotlin.time.Duration
+
+
+@Inject(true, 1)
+class ReminderService {
+
+    private val reminderManager by KoinUtil.inject<ReminderManager>()
+    private val missingUserManager by KoinUtil.inject<MissingUserManager>()
+    private val shardManager by KoinUtil.inject<ShardManager>()
+    private val logger by Log
+
+    /** it's the job of [ReminderCommand] to cancel [waitingJob] */
+    var waitingJob = TaskScope.launch { }
+
+    init {
+        pollingLoop()
+    }
+
+    private fun pollingLoop() = TaskScope.launch {
+        while (true) {
+            val reminderEntry = reminderManager.getNextUpcomingReminder()
+            if (reminderEntry == null) {
+                waitingJob = launch { delay(Duration.INFINITE) }
+                waitingJob.join()
```

Comment:
Because this is nontrivial, perhaps some documentation (a comment) explaining that this task will be interrupted is in order